keps/sig-apps/3998-job-success-completion-policy/README.md

```diff
@@ -0,0 +1,924 @@
+The default behavior will be applied to running workloads.
+
+###### What specific metrics should inform a rollback?
+
+An increase in the `job_sync_duration_seconds` metrics can mean that finished jobs
+are taking longer to evaluate.
+
+The users should check whether the completed jobs have the appropriate condition,
+specifically the reason.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+No.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+`job_finished_total`: the new `JobSuccessPolicy` and `Completions` values for the `reason` label. 
+  - `JobSuccessPolicy` indicates a job is declared as succeeded because the job meets `.spec.successPolicy`.
+  - `Completions` indicates a job is declared as succeeded because the job meets `spec.completions`.
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [x] Job API .status
+  - The Job controller will add a condition with `JobSuccessPolicy` reason to `conditions`.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+99% percentile over day for Job syncs is <= 15s for a client-side 50 QPS limit.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [x] Metrics
+  - Metric name: `job_sync_duration_seconds` (existing): can be used to see how much the
+    feature enablement increases the time spent in the sync job
+  - Components exposing the metric:
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+No.
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No.
+
+### Scalability
+
+No.
+
+###### Will enabling / using this feature result in any new API calls?
+
+No.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+Yes, it will increase the size of existing API objects only when the `.spec.successPolicy` is set.
+
+- API type(s): Job
+- Estimated increase in size: `.spec.successPolicy.criteria.succeededIndexes` field are impacted.
+In the worst case, the size of `succeededIndexes` can be estimated about 64KiB (see [Risks and Mitigations](#risks-and-mitigations)).
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No.
```

Comment:
I would say that more CPU is posible because computing if a given set of indexes completed (especially if there are many of them - approaching 64kB) may actually be non-negligible.
Thoughts?

Reply:
Great point!
Since `SuccessPolicy` uses compressed indexes in the `successPolicy.criteria[*].succeededIndexes`, the job controller needs more computing resources in the case of many indexes.